packages/application-extension/src/index.ts

```diff
@@ -254,33 +254,37 @@ function addCommands(app: JupyterLab, palette: ICommandPalette): void {
 
   command = CommandIDs.toggleLeftArea;
   app.commands.addCommand(command, {
-    label: 'Show Left Area',
+    label: args => args['isPalette'] ?
+    'Toggle Left Area' : 'Show Left Area',
```

Comment:
I wonder whether it would be better for this to just be `Hide Left Area`, since the only time you can see the item is when it is shown.

Furthermore, we have noticed that phosphor does not render keyboard shortcuts if the `args` do not match. In practice, this means that the `isPalette` trick prevents the shortcut from showing in the command palette here. That may be out of the scope of this PR, though.

Reply:
> Furthermore, we have noticed that phosphor does not render keyboard shortcuts if the args do not match. In practice, this means that the isPalette trick prevents the shortcut from showing in the command palette here. That may be out of the scope of this PR, though.

Interesting!

Reply:
I guess it makes sense? The keyboard shortcut is for very specific args...

Reply:
I guess a delegate command would also have a problem showing the keyboard shortcut.

If the command palette showed the toggle state, I think my concern would be resolved on that point.

Reply:
Why would a delegate command have a problem? They don't really use the `args` much.

Reply:
The keyboard shortcut is tied to the original command, not the delegate, right? So the keyboard shortcut won't show up for the delegate command.

Reply:
No, the shortcut is tied to the delegate. In some cases the same shortcut is registered to both, and the tie goes to the more specific CSS selector.

Reply:
Sorry, I think I should have said "Proxy command" rather than delegate. If you make a new "proxy" command with the right label that just calls the real command, and add the proxy command to the command palette, the keyboard shortcut for the underlying command still won't show up.

Reply:
Ah, I see. I thought you were referring to the delegators for the semantic menu items. Yes, you are right, I think.